File: loggalitic/src/main/java/com/android/oleksandrpriadko/loggalitic/analytics/AnalyticsEvent.java
package com.android.oleksandrpriadko.loggalitic.analytics;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.text.TextUtils;

import static com.google.common.base.Preconditions.checkArgument;

/**
 * Generic event which have to be converted to necessary type.
 */
public class AnalyticsEvent {

    private static final String DESCRIPTION = "description";

    private String mName;
    private final Bundle mAttributes = new Bundle();

    public AnalyticsEvent(@NonNull String name) {
        checkArgument(!TextUtils.isEmpty(name));
        this.mName = name;
    }

    public AnalyticsEvent(String name, String description) {
        this(name);
        this.mAttributes.putString(DESCRIPTION, description);
    }


    public String getName() {
        return mName;
    }

    public Bundle getAttributes() {
        return mAttributes;
    }
}

File: loggalitic/src/main/java/com/android/oleksandrpriadko/loggalitic/analytics/AnalyticsEvent.kt
package com.android.oleksandrpriadko.loggalitic.analytics

import android.os.Bundle
import android.text.TextUtils

import com.google.common.base.Preconditions.checkArgument

/**
 * Generic event which have to be converted to necessary type.
 */
class AnalyticsEvent(val name: String) {
    val attributes = Bundle()

    init {
        checkArgument(!TextUtils.isEmpty(name))
    }

    constructor(name: String, description: String) : this(name) {
        this.attributes.putString(DESCRIPTION, description)
    }

    companion object {

        private const val DESCRIPTION = "description"
    }
}
